### DISCUSSION TITLE
javasdk的search余弦相似度查不出数据，是为什么

### DISCUSSION THREAD

**Discussion by blessing59420** [milvus-io/milvus#27524] [Q&A and General discussion]
public List<Long> serarchSimilar(String msg){

        EmbeddingApiResponse embedding = embeddingModel.doEmbedding(msg);
        if (embedding == null || embedding.getStatus()!=200) {
            log.error("请求embedding失败，为空");
            return null;
        }

        List<List<Float>> vectors = new ArrayList<>();
        vectors.add(embedding.getData().getVector());
        log.info("搜索：{}",JSONUtil.toJsonStr(vectors));
        final String SEARCH_PARAM = "{\"nprobe\":10,\"offset\":1, \"limit\":3}";    // Params
        SearchParam param = SearchParam.newBuilder()
                .withCollectionName(collectionName)
                .withMetricType(MetricType.COSINE)  // 使用余弦相似度
                .withTopK(10)
                .withVectors(vectors)  // 使用新的vectors变量
                .withVectorFieldName("vector")
                .withConsistencyLevel(ConsistencyLevelEnum.EVENTUALLY)
                .withParams(SEARCH_PARAM)
                .build();
        System.out.println("请求参数："+JSONUtil.toJsonStr(param));

        R<SearchResults> response = milvusClient.search(param);
        if (response.getStatus() != R.Status.Success.getCode()) {
            System.out.println(response.getMessage());
            return null;
        }
        System.out.println("返回："+JSONUtil.toJsonStr(response));

        SearchResultsWrapper wrapper = new SearchResultsWrapper(response.getData().getResults());
        log.info("结果:{}",JSONUtil.toJsonStr(wrapper));
        return  null;
    }

这是我的代码，，但是查询一直返回，为什么：
{"status":0,"data":{"unknownFields":{}}}

我的go代码就没问题可以查出来，javasdk是 余弦相似度或者IP 都查不出来数据，这是哪里有问题呢

**Comment by blessing59420** [comment]
<img width="1103" alt="image" src="https://github.com/milvus-io/milvus/assets/59166341/12f34196-9b6d-442e-bfb6-f61d1fc6e395">
我的go代码就没有问题，就是java这边一直查不出，怎么解决

**Reply by yelusion2** [reply]
@blessing59420 how many result this search will be return?  i see u use python sdk the param 'offset' value is 0 and the java sdk value is 1. u can change the 'offset' value to 0 using java sdk try again?
